pkg/install/stack/common/validation.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+)
+
+func ValidateRandomSecret(value string, path string) error {
+	if value == "" {
+		secret, err := randomString()
+		if err == nil {
+			return fmt.Errorf("%s must be a non-empty secret, for example: %s", path, secret)
+		}
+
+		return fmt.Errorf("%s must be a non-empty secret", path)
+	}
+
+	return nil
+}
```

Comment:
What is this supposed to do? If the value is not present, we are going to return an error in any case. Why do we generate a random string?

Reply:
To be nice to the user. To prevent us from having to formulate an awkward "Well you can use this random combination of shell utilities, but only on this particular Debian derivate. On Mac use this, and on Windows, oh boy, good luck. Make sure that the secret is between X and Y bytes long, contains at least 3 special characters and is also a palindrome when read in Hebrew." recommendation. Instead, if the user did not configure a secret, we offer them a strong random one right there for copy&pasting.

Reply:
https://github.com/kubermatic/kubermatic/pull/6435#discussion_r571940212